test/avrora/encoder_test.exs

```diff
@@ -402,6 +402,53 @@ defmodule Avrora.EncoderTest do
     end
   end
 
+  describe "decode_plain/2" do
+    test "when decoding plain message that starts with what looks like a magic byte" do
+      schema_name = "io.confluent.numeric_transfer"
+      numeric_transfer_schema = numeric_transfer_schema()
+
+      Avrora.Storage.MemoryMock
+      |> expect(:get, fn key ->
+        assert key == "io.confluent.numeric_transfer"
+
+        {:ok, nil}
+      end)
+      |> expect(:put, fn key, value ->
+        assert key == "io.confluent.numeric_transfer"
+        assert value == numeric_transfer_schema
+
+        {:ok, value}
+      end)
+
+      Avrora.Storage.RegistryMock
+      |> expect(:put, fn key, value ->
+        assert key == "io.confluent.numeric_transfer"
+        assert value == numeric_transfer_json_schema()
+
+        {:error, :unconfigured_registry_url}
+      end)
+
+      Avrora.Storage.FileMock
+      |> expect(:get, fn key ->
+        assert key == "io.confluent.numeric_transfer"
+
+        {:ok, numeric_transfer_schema}
+      end)
+
+      assert {:ok, decoded} =
+               Avrora.decode_plain(
+                 numeric_transfer_plain_message_0(),
+                 schema_name: schema_name
+               )
```

Comment:
I think this assert could be just an assignment statement because it's not validating match

```suggestion
      {:ok, decoded} =
        Avrora.decode_plain(numeric_transfer_plain_message_fake_magic_byte(), schema_name: schema_name)
```